### PR TITLE
Calculates AC

### LINKED ISSUE #41: Create a AC, for creature on import

### PR DESCRIPTION
**Pull request by morkusporkus**
Closes #41

### CODE REVIEW

File: src/Core/Models/Imports/TetraCubeCreature.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
+﻿using System.Text.Json.Serialization;
+
+namespace DMPowerTools.Core.Models.Imports
+{
+    public class TetraCubeCreature
+    {
+        public int Id { get; set; }
```
Comment (benjaminsampica) [inline src/Core/Models/Imports/TetraCubeCreature.cs:7]: What’s their Id and do we care

File: src/Core/Models/Imports/TetraCubeCreature.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        public int FlySpeed { get; set; }
+        public bool Hover { get; set; }
+        public int SwimSpeed { get; set; }
+        public bool CustomHP { get; set; }
+        public bool CustomSpeed { get; set; }
+        public string HpText { get; set; }
+        public string SpeedDesc { get; set; }
+        public int StrPoints { get; set; }
+        public int DexPoints { get; set; }
+        public int DexModifier
```
Comment (benjaminsampica) [inline src/Core/Models/Imports/TetraCubeCreature.cs:30]: This should probably live in somewhere else as it’s the calculation for 5e in general.

File: src/Core/Models/Imports/TetraCubeCreature.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        public bool DoubleColumns { get; set; }
+        public int SeparationPoint { get; set; }
+
+    }
+    public static class TetraCubeCreatureExtensions
+    {
+        public static int CalculateACFromTetraCube(this TetraCubeCreature creature)
+        {
+            var modifierWithCap = creature.DexModifier;
+            if (modifierWithCap > 2)
```
Comment (benjaminsampica) [inline src/Core/Models/Imports/TetraCubeCreature.cs:78]: Regarding the 5e calc from above I think this is the same thing.

File: src/Maui/Features/Creatures/Import/Upload.razor.cs [hunk 4]
```diff
@@ -11,11 +13,14 @@ private async Task UploadFilesAsync(IReadOnlyList<IBrowserFile> files)
     {
         foreach (var file in files)
         {
-            var creature = await JsonSerializer.DeserializeAsync<Creature>(file.OpenReadStream(512000, default), new JsonSerializerOptions
+            var tetraCubeCreature = await JsonSerializer.DeserializeAsync<TetraCubeCreature>(file.OpenReadStream(512000, default), new JsonSerializerOptions
             {
                 PropertyNameCaseInsensitive = true
             });
-
+            var config = new MapperConfiguration(cfg => cfg.CreateMap<TetraCubeCreature, Creature>());
```
Comment (benjaminsampica) [inline src/Maui/Features/Creatures/Import/Upload.razor.cs:20]: We’re configuring this through reflection. Look in Core for some profile examples.

File: src/Tests/Extensions/TetraCubeCreatureExtensionTests.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public class TetraCubeCreatureExtensionTests : IntegrationTestBase
+    {
+
+        [Fact]
+        public async Task GivenATetraCubeCreature_WhenCreatingACreature_ThenCalculateArmorClass()
+        {
+            var mockTetraCubeCreature = CreateFakeTetraCubeCreature();
+            mockTetraCubeCreature.ArmorName = "padded";
+            var expectedResult = 11 + mockTetraCubeCreature.DexModifier + mockTetraCubeCreature.ShieldBonus;
```
Comment (benjaminsampica) [inline src/Tests/Extensions/TetraCubeCreatureExtensionTests.cs:13]: Like above comments Should probably be somewhere since it’s a core calculation.

**Review by benjaminsampica — CHANGES_REQUESTED** [review]

File: src/Core/Models/Imports/TetraCubeCreature.cs (hunk 1 above)
Comment (morkusporkus) [reply src/Core/Models/Imports/TetraCubeCreature.cs:7]: no we dont

File: src/Core/Models/Imports/TetraCubeCreature.cs (hunk 3 above)
Comment (morkusporkus) [reply src/Core/Models/Imports/TetraCubeCreature.cs:78]: hmm I agree with the other one, not sure if I agree that modifer is a core calculation or calculate from tetracube is

Comment (benjaminsampica) [reply src/Core/Models/Imports/TetraCubeCreature.cs:78]: Isn’t this the medium armor maximum?

File: src/Maui/Startup.cs [hunk 6]
```diff
@@ -21,6 +21,7 @@ public void ConfigureServices(IServiceCollection services)
         services.AddMudServices();
         services.AddMediatR(typeof(ManageCombatQueryHandler));
         services.AddAutoMapper(typeof(ManageCombatQueryResponse));
+        services.AddAutoMapper(typeof(TetraCubeCreature));
```
Comment (benjaminsampica) [inline src/Maui/Startup.cs:24]: AddAutomapper does assembly scanning. You don’t need this more than once.

File: src/Core/DMPowerTools.Core.csproj [hunk 7]
```diff
@@ -12,6 +12,7 @@
 	  <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="12.0.0" />
 	  <PackageReference Include="MediatR" Version="11.1.0" />
 	  <PackageReference Include="MediatR.Extensions.Microsoft.DependencyInjection" Version="11.0.0" />
+	  <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="7.0.1" />
```
Comment (benjaminsampica) [inline src/Core/DMPowerTools.Core.csproj:15]: What’s this package needed for?

File: src/Core/Features/Creatures/Import.cs [hunk 8]
```diff
@@ -46,6 +50,45 @@ public async Task<Unit> Handle(AcceptAllCreaturesCommand request, CancellationTo
...
+}
+
+public record ProcessMonsterFileCommandResponse(List<Creature> Creatures) { }
+
+public class ProcessMonsterFileProfile : Profile
+{
+    public ProcessMonsterFileProfile()
+    {
+        CreateMap<TetraCubeCreature, Creature>()
+            .ForMember(c => c.Id, tcc => tcc.Ignore());
```
Comment (benjaminsampica) [inline src/Core/Features/Creatures/Import.cs:89]: There is no property to ignore here. Do we need to?

**Review by benjaminsampica — APPROVED** [review]

**Pull request merged**